lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -957,23 +997,39 @@ LowererMDArch::LowerCall(IR::Instr * callInstr, uint32 argCount)
     // Manually home args
     if (shouldHomeParams)
     {
-        static const RegNum s_argRegs[IntArgRegsCount] = {
-    #define REG_INT_ARG(Index, Name)  Reg ## Name,
-    #include "RegList.h"
-        };
-
         const int callArgCount = this->helperCallArgsCount + static_cast<int>(argCount);
-        const int argRegs = min(callArgCount, static_cast<int>(IntArgRegsCount));
-        for (int i = argRegs - 1; i >= 0; i--)
+        const int xpTopPos = this->xplatCallArgs.GetTopPosition();
+
+        int floatCount = this->xplatCallArgs.floatCount;
+        int argRes = min(callArgCount, static_cast<int>(XmmArgRegsCount));
+        int intCount = max(0, min(argRes - floatCount, static_cast<int>(IntArgRegsCount)));
+        int argRegs = intCount + floatCount;
```

Comment:
Assuming you can guarantee that the partial recording successfully records every float register arg, the logic here seems quite complex and could contain bugs. E.g., given 8 args: `int0, int1, int2, int3, int4, int5, int_stack, xmm7`.
```
callArgCount == 8
xpTopPos == 8
floatCount == 1
argRes == 8
intCount == 6
argRegs == 7
```

Then your following `for` loop will start from `i = 8` and end at `i == 2` (loop 7 iterations because of argRegs). You'll miss homing the 1st register `int0`.

Seems it could be simpler:
```
for (int i = min(callArgCount, XmmArgRegsCount); i > 0; i--) {
  if (had recording for i and it is float) {
    prep homing float reg
  } else if (i <= IntArgRegsCount) {
    prep homing int reg
  }
  ...
}
```